transformer_engine/debug/features/api.py

```diff
@@ -133,9 +145,14 @@ def modify_tensor_enabled(
         gemm: str,
         tensor_name: str,
         iteration: int,
-    ) -> bool:
+    ) -> bool | Tuple[bool, int | float]:
```

Comment:
Why imply inconsistent output types?
```suggestion
    ) -> Tuple[bool, Optional[int]]:
```
This just makes them much more awkward to use, i.e. we need to put `_as_pair` everywhere. I can see two cases:
- We want to enforce the same API for functions that return iter info and functions that don't. But what benefit does that bring? Better to have different APIs and only do iter-related logic when it's actually needed.
- We want to ensure backward compatibility, e.g. with subclasses implemented outside of TE. Perhaps it would be better to create a new API for functions that return iter info?